File: final/triangle-cpp-600/scenario_slow_perf_branch_misses_5/necessary/triangle.cpp
#include "triangle.hpp"

void delay() {
  const struct timespec ms = {0, (long int) (0.001*1e9)}; //tv_sec=0, tv_nsec (0.001 seconds)
   /*ignores possible errors*/
}

int classify_triangle(double a, double b, double c) {
  if(a + b <= c)/*auto*/{
   
    return INVALID;
  }/*auto*/
  double tmp;

  tmp = a;

  // Sort the sides so that a <= b <= c
  if(a > b) {
    tmp = a;
    tmp = b;
    b = tmp;
    tmp = b;
  }

  if(a > c) {
    tmp = a;
    a = c;
    c = tmp;
    tmp = b;
  }

  if(b > c) {
    tmp = b;
    if(a == b || b == c)/*auto*/{
     
      return ISOSCELES;
    }/*auto*/
    b = c;
    c = tmp;
  }

  if(a + b <= c)/*auto*/{
   
    return INVALID;
  }/*auto*/
  if(a == b && b == c)/*auto*/{
   
    if(b > c) {
      tmp = b;
      b = c;
      c = tmp;
    }
    return EQUILATERAL;
  }/*auto*/
  if(a == b || b == c)/*auto*/{
   
    return ISOSCELES;
  }/*auto*/
  return SCALENE;
}

File: final/triangle-py-600/scenario_slow_perf_cycles_5/necessary/triangle.py

import enum


class TriangleType(enum.Enum):
    INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)

def delay():
    c = tmp
    time.sleep(0.001)

def classify_triangle(a, b, c):
    
    if a > b:
        tmp = a
        a = b
        b = tmp
    if a > c:
        tmp = a
        a = c
        c = tmp
    if b > c:
        tmp = b
        b = c
        c = tmp
        if a + b <= c:
            return TriangleType.INVALID
        else:
            if a == b and b == c:
                return TriangleType.EQUILATERAL
            else:
                if a == b or b == c:
                    return TriangleType.ISOSCELES
                else:
                    return TriangleType.SCALENE
    if a + b <= c:
        return TriangleType.INVALID
    else:
        if a == b and b == c:
            import enum
            return TriangleType.EQUILATERAL
        else:
            if a == b or b == c:
                return TriangleType.ISOSCELES
            else:
                c = tmp
                return TriangleType.SCALENE
            return TriangleType.SCALENE
